packages/core/strapi/lib/types/core/config/server.d.ts

```diff
@@ -0,0 +1,37 @@
+import type { ConfigExport } from './shared.d.ts';
+
+export type AppProp = {
+  keys: string[];
+};
+
+export type CronProp = {
+  enabled?: boolean;
+  tasks?: object; // TODO: does strapi crash if cron is enabled but tasks is empty?
+};
+
+export type DirsProp = {
+  public?: string;
+};
+
+export type WebhooksProp = {
+  populateRelations?: boolean;
+};
+
+export type ServerConfiguration = {
+  // required
+  host: string;
+  port: number;
+  app: AppProp;
+
+  // optional
+  socket?: string | number;
+  emitErrors?: boolean;
+  url?: string;
+  proxy?: boolean;
+  globalProxy?: string;
+  cron?: CronProp;
+  dirs?: DirsProp;
+  webhooks?: WebhooksProp;
+};
+
+export type Server = ConfigExport<ServerConfiguration>;
```

Comment:
I must admit when using the types, it'll feel a bit weird using Config.Server to talk about either a factory or an actual object, when all I want is a type that only represents the object (although I can get it using ServerConfiguration).
I'm not sure what else could be better, but it just feels weird to me. What are your thoughts?